spacy/kb/candidate.pyx

```diff
@@ -1,74 +1,116 @@
+        raise NotImplementedError
 
     @property
-    def entity_freq(self) -> float:
-        return self.entity_freq
-
-    @property
-    def entity_vector(self) -> Iterable[float]:
-        return self.entity_vector
+    def entity_vector(self) -> vector[float]:
+        """RETURNS (vector[float]): Entity vector."""
+        return self._entity_vector
 
     @property
     def prior_prob(self) -> float:
-        return self.prior_prob
+        """RETURNS (List[float]): Entity vector."""
+        return self._prior_prob
 
 
-def get_candidates(kb: KnowledgeBase, mention: Span) -> Iterable[Candidate]:
-    """
-    Return candidate entities for a given mention and fetching appropriate entries from the index.
-    kb (KnowledgeBase): Knowledge base to query.
-    mention (Span): Entity mention for which to identify candidates.
-    RETURNS (Iterable[Candidate]): Identified candidates.
-    """
-    return kb.get_candidates(mention)
+cdef class InMemoryCandidate(Candidate):
+    """Candidate for InMemoryLookupKB."""
 
+    def __init__(
+        self,
+        kb: InMemoryLookupKB,
+        entity_hash: int,
+        mention_hash: int,
+        entity_vector: vector[float],
+        prior_prob: float,
+        entity_freq: float
+    ):
+        """
+        kb (InMemoryLookupKB]): InMemoryLookupKB instance.
+        entity_id (int): Entity ID as hash that can be looked up with InMemoryKB.vocab.strings.__getitem__().
+        entity_freq (int): Entity frequency in KB corpus.
+        entity_vector (List[float]): Entity embedding.
+        mention_hash (int): Mention hash.
+        prior_prob (float): Prior probability of entity for this mention - i.e. the probability that, independent of
+            the context, this mention resolves to this entity_id in the corpus used to build the knowledge base. In
+            cases in which this isn't always possible (e.g.: the corpus to analyse contains mentions that the KB corpus
+            doesn't) it might be better to eschew this information and always supply the same value.
+        """
+        super().__init__(
+            entity_id=kb.vocab.strings[entity_hash],
+            entity_vector=entity_vector,
+            prior_prob=prior_prob,
+        )
+        self._kb = kb
+        self._mention = mention_hash
+        self._entity_id = entity_hash
+        self._entity_freq = entity_freq
 
-def get_candidates_batch(kb: KnowledgeBase, mentions: Iterable[Span]) -> Iterable[Iterable[Candidate]]:
-    """
-    Return candidate entities for the given mentions and fetching appropriate entries from the index.
-    kb (KnowledgeBase): Knowledge base to query.
-    mention (Iterable[Span]): Entity mentions for which to identify candidates.
-    RETURNS (Iterable[Iterable[Candidate]]): Identified candidates.
-    """
-    return kb.get_candidates_batch(mentions)
+    @property
+    def mention(self) -> str:
+        """RETURNS (str): ID/name of this entity in the KB"""
```

Comment:
update doc string

Reply:
Updated in https://github.com/explosion/spaCy/pull/12341/commits/348dd1c87ec853f837907ef8b804b103be1ebfcc.